src/Cowsay.zig

```diff
@@ -55,85 +53,67 @@ fn print(self: *Self, comptime fmt: []const u8, comptime args: anytype) !void {
 }
 
 fn printHLine(self: *Self) !void {
-    try self.w.writeByte('+');
-    try self.w.writeByteNTimes('-', self.max_line_length + 2);
-    try self.w.writeAll("+\n");
+    try self.writer.writeByte('+');
+    try self.writer.writeByteNTimes('-', self.max_line_length + 2);
+    try self.writer.writeAll("+\n");
 }
 
-fn printMessage(self: *Self, s: []const u8) !void {
-    var line_len: u32 = 0;
-    for (s) |c| {
-        if (line_len == 0) {
-            try self.w.writeAll("| ");
-        }
-        if (c == '\n') {
-            try self.w.writeByteNTimes(' ', self.max_line_length - line_len);
-            try self.w.writeAll(" |");
-            line_len = 0;
-        } else {
-            line_len += 1;
-        }
-        try self.w.writeByte(c);
-    }
-    if (line_len != 0) {
-        try self.w.writeByteNTimes(' ', self.max_line_length - line_len);
-        try self.w.writeAll(" |\n");
+fn printMessage(self: Self, s: []const u8) !void {
+    var lines = std.mem.splitScalar(u8, s, '\n');
```

Comment:
Instead of parsing character by character (unless there is an inner parsing loop that _must_ be extremely optimized), it is typically preferable to just use the standard library's `split` functions.

Reply:
Was looking for a split function! Did not think it will be in `std.mem`!